src/components/application_manager/test/commands/mobile/set_app_icon_test.cc

```diff
@@ -144,6 +175,339 @@ TEST_F(SetAppIconRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
   }
 }
 
+TEST_F(SetAppIconRequestTest, OnEvent_InvalidEnum_ReceivedUnknownEvent) {
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  Event event(hmi_apis::FunctionID::INVALID_ENUM);
+  event.set_smart_object(*msg);
+
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _)).Times(0);
+
+  request_->on_event(event);
+}
+
+TEST_F(SetAppIconRequestTest, OnEvent_NonExistentApp_NullPointer) {
+  MockAppPtr mock_app_nonexistent;
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_nonexistent));
+
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::UNSUPPORTED_RESOURCE;
+  (*msg)[am::strings::msg_params][am::strings::info] = "info1";
+
+  Event event(hmi_apis::FunctionID::UI_SetAppIcon);
+  event.set_smart_object(*msg);
+
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _)).Times(0);
+
+  request_->on_event(event);
+}
+
+TEST_F(SetAppIconRequestTest, Run_AllCorrectData_SendHMIRequest) {
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .Times(2)
+      .WillRepeatedly(Return(mock_app_));
+
+  MockProtocolHandler mock_app_protocol_handler;
+  EXPECT_CALL(app_mngr_, protocol_handler())
+      .WillOnce(ReturnRef(mock_app_protocol_handler));
+
+  MockProtocolHandlerSettings mock_protocol_handler_settings;
+  EXPECT_CALL(mock_app_protocol_handler, get_settings())
+      .WillRepeatedly(ReturnRef(mock_protocol_handler_settings));
+
+  const protocol_handler::MajorProtocolVersion max_supported_protocol_version =
+      protocol_handler::PROTOCOL_VERSION_4;
+  EXPECT_CALL(mock_protocol_handler_settings, max_supported_protocol_version())
+      .WillRepeatedly(Return(max_supported_protocol_version));
+
+  const uint32_t storage_max_size = 10u;
+  EXPECT_CALL(app_mngr_settings_, app_icons_folder_max_size())
+      .WillRepeatedly(ReturnRef(storage_max_size));
+
+  const std::string app_policy_id = "test_icon";
+  EXPECT_CALL(*mock_app_, policy_app_id())
+      .WillRepeatedly(Return(app_policy_id));
+
+  EXPECT_CALL(*mock_app_, app_id()).WillRepeatedly(Return(kAppId));
+
+  MessageSharedPtr result;
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  ASSERT_TRUE(request_->Init());
+  request_->Run();
+
+  const std::string expected_sync_file_name = file_system::ConvertPathForURL(
+      kAppStorageFolder + "/" + kAppFolder + "/" + kSyncFileName);
```

Comment:
@BSolonenko Can we use global variables that you defined?

Reply:
@AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/1974/commits/ce3bce3dc25e2692cf0256f39dffb6b98effa327